### DISCUSSION TITLE
Can't get metro-resolver-symlinks to work

### DISCUSSION THREAD

**Discussion by donaldkwong** [microsoft/rnx-kit#2059] [Q&A]
I'm having trouble getting metro-resolver-symlinks to work with importing modules in my monorepo with symlinks. My RN app is currently in `/apps/app1` and I want to import a module in `/modules/module1`. I ran `npm i ../../modules/module1` in the app directory and the `package.json` and `package-lock.json` files have been updated with the new dependency. I also see a symlink created in the `node_modules` directory. I can even add an import statement in my .tsx file and pull in some symbols to use in my code. I followed the directions and made the corresponding changes to `metro.config.js`. However, when I try to run the app (with `npm run ios`), metro complains that the module can't be resolved. Am I doing something wrong?

**Comment by tido64** [comment]
Hi @donaldkwong, do you have a repro we can have a look at? What does your `metro.config.js` look like? Did you add an entry for the `modules` folder in your config, under `watchFolders`?

**Comment by donaldkwong** [comment]
Hey, thanks for the quick response. I don't have an easy repro, unfortunately. However, adding the folders to `watchFolders` did work. I was under the assumption that I wouldn't have to keep that updated if I used `metro-resolver-symlinks`. Is that not the case?

**Reply by tido64** [reply]
Metro only interfaces with an internal mapping of the file system. This mapping is provided by a file watcher, like Watchman. For a package to be correctly resolved, it must also be watched. This is a Metro implementation detail that we cannot get around (yet).

**Comment by donaldkwong** [comment]
FWIW, the documentation [here](https://microsoft.github.io/rnx-kit/docs/guides/bundling) implies that `watchFolders` doesn't need to be specified anymore:

<img width="885" alt="image" src="https://user-images.githubusercontent.com/866083/211087456-db25bad5-8574-4608-89db-36ca3677fe00.png">
